src/Search.php

```diff
@@ -727,85 +727,21 @@ protected function set_general_context( &$context ) {
 	 * @throws UnexpectedValueException When filter type is not recognized.
 	 */
 	protected function set_filters_context( &$context ) {
-		// Retrieve P4 settings in order to check that we add only categories that are children of the Issues category.
-		$options = get_option( 'planet4_options' );
+		// Categories.
+		$context['categories'] = Search\Filters\Categories::get_filters();
+		uasort( $context['categories'], fn ( $a, $b ) => strcmp( $a['name'], $b['name'] ) );
 
-		// Category <-> Issue.
-		// Consider Issues that have multiple Categories.
-		$categories = get_categories( [ 'child_of' => $options['issues_parent_category'] ] );
-		if ( $categories ) {
-			foreach ( $categories as $category ) {
-				$context['categories'][ $category->term_id ] = [
-					'term_id' => $category->term_id,
-					'name'    => $category->name,
-					'results' => 0,
-				];
-			}
-		}
+		// Post Types.
+		$context['post_types'] = Search\Filters\PostTypes::get_filters();
 
-		// Tag <-> Campaign.
-		$tags = get_terms(
-			[
-				'taxonomy'   => 'post_tag',
-				'hide_empty' => false,
-			]
+		// Content Types.
+		$context['content_types'] = Search\Filters\ContentTypes::get_filters(
+			self::should_include_archive()
```

Comment:
We could include the check whether it should include the archive inside of `get_filters`. I assume that if it would be called from another place, it would need the same value anywhere?

Reply:
I don't know, that's why I made a parameter of it :grimacing:  if I move it to the filter, should I reverse the usage of this value in the Search class (use the filter method to define the archive usage), or keep a duplicate method in Search and Filter ?